### PR TITLE
[Merged by Bors] - feat(Combinatorics/SimpleGraph): darts and edges are injective for walks, length of paths and trails are bounded

### PR DESCRIPTION
**Pull request by henkkuli**
Provide theorems stating that `SimpleGraph.Walk.darts` and `SimpleGraph.Walk.edges` are injective functions, and that the length of `SimpleGraph.Walk.IsTrail` and `SimpleGraph.Walk.IsPath` is bounded for finite graphs.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Path.lean
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Path.lean:190]:
```suggestion
    apply w.edges_subset_edgeSet
```

File: Mathlib/Combinatorics/SimpleGraph/Path.lean
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Path.lean:238]:
```suggestion
  exact Finset.card_le_card support.subset_univ
```

File: Mathlib/Combinatorics/SimpleGraph/Walk.lean
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Walk.lean:717]:
```suggestion
  | .nil, .nil, _ => rfl
```

File: Mathlib/Combinatorics/SimpleGraph/Path.lean
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Path.lean:182]:
```suggestion
    {w : G.Walk u v} (h : w.IsTrail) : w.length ≤ G.edgeFinset.card := by
```

File: Mathlib/Combinatorics/SimpleGraph/Path.lean
Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/Path.lean:230]: `IsPath.length_lt_card_verts`

**Comment by henkkuli** [comment]
Thanks for the feedback! I applied the suggested changes.

Turns out that `IsPath.length_lt_card_V` already exists as `IsPath.length_lt`, so I removed that theorem. It was a last-minute change for symmetry with `IsTrail.length_le_card_edgeFinset`, and apparently I didn't do my due diligence well enough.

**Comment by henkkuli** [comment]
Is there still something I should do to get this merged, or is this just waiting for a second round of reviews?

**Comment by IvanRenison** [comment]
I'm not a maintainer, so I can't marge it, a maintainer has to merge it, so it's waiting. Any way, you should mark as resolved the suggestions that made.

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Review by bryangingechen — APPROVED** [review]
Thanks!
bors r+

**Pull request closed without merging**
